mir_dataset_loaders/salami.py

```diff
@@ -0,0 +1,185 @@
+                  any(SALAMI_INDEX[track_id]['annotator_2_uppercase'])]
+    sections_data = _load_sections(get_local_path(annotations_dir, track_id), annotators)
+
+    return SalamiTrack(
+        track_id,
+        sections_data[0],
+        sections_data[1],
+        sections_data[2],
+        sections_data[3],
+        track_metadata['source'],
+        track_metadata['annotator_1_id'],
+        track_metadata['annotator_2_id'],
+        track_metadata['duration_sec'],
+        track_metadata['title'],
+        track_metadata['artist'],
+        track_metadata['annotator_1_time'],
+        track_metadata['annotator_2_time'],
+        track_metadata['class'],
+        track_metadata['genre']
+    )
+
+
+def _load_sections(sections_path, annotators):
+    sections_data = []
+    for a in range(len(annotators)):
+        times, secs = [], []
+        for f in ['uppercase.txt', 'lowercase.txt']:
+            if annotators[a]:
+                file_path = os.path.join(sections_path, 'parsed', 'textfile'+str(a+1)+'_'+f)
+                with open(file_path, 'r') as fhandle:
+                    reader = csv.reader(fhandle, delimiter='\t')
+                    for line in reader:
+                        times.append(float(line[0]))
+                        secs.append(line[1])
+
+                sections_data.append(SectionsData(np.array(times)[:-1], np.array(times)[1:], np.array(secs)))
+
+            else:
+                times, secs = None, None
+                sections_data.append(None)
+
+    return sections_data
+
+
+def _load_metadata(data_home):
+
+    metadata_relative_path = os.path.join('salami-data-public-master','metadata', 'metadata.csv')
+    metadata_path = get_local_path(
+        data_home, metadata_relative_path)
+
+    if not os.path.exists(metadata_path):
+        return None
+
+    with open(metadata_path, 'r') as fhandle:
+        reader = csv.reader(fhandle, delimiter=',')
+        raw_data = []
+        for line in reader:
+            if line[0] == 'SONG ID':
+                continue
+            raw_data.append(line)
+
+    metadata_index = {}
+    for line in raw_data:
+        track_id = line[0]
+
+        metadata_index[track_id] = {
+            'source': line[1],
+            'annotator_1_id': line[2],
+            'annotator_2_id': line[3],
+            'duration_sec': line[5],
+            'title': line[7],
+            'artist': line[8],
+            'annotator_1_time': line[10],
+            'annotator_2_time': line[11],
+            'class': line[14],
+            'genre': line[15],
+        }
+
+    metadata_index['data_home'] = data_home
+
+    return metadata_index
+
+
+def _reload_metadata(data_home):
+    global SALAMI_METADATA
+    SALAMI_METADATA = _load_metadata(data_home=data_home)
+
+
+def cite():
+    cite_data = """
+===========  MLA ===========
+Smith, Jordan Bennett Louis, et al. 
+"Design and creation of a large-scale database of structural annotations." 
+ISMIR. Vol. 11. 2011.
```

Comment:
I think it's not the best citation - i think it's treating ISMIR as a journal not a conference.